eth/beacon/types/validator_records.py

```diff
@@ -65,28 +66,30 @@ def __init__(self,
             withdrawal_credentials=withdrawal_credentials,
             randao_commitment=randao_commitment,
             randao_layers=randao_layers,
-            status=status,
-            latest_status_change_slot=latest_status_change_slot,
+            activation_slot=activation_slot,
+            exit_slot=exit_slot,
+            withdrawal_slot=withdrawal_slot,
+            penalized_slot=penalized_slot,
             exit_count=exit_count,
+            status_flags=status_flags,
             custody_commitment=custody_commitment,
             latest_custody_reseed_slot=latest_custody_reseed_slot,
             penultimate_custody_reseed_slot=penultimate_custody_reseed_slot,
         )
 
-    @property
-    def is_active(self) -> bool:
+    def is_active(self, slot: int) -> bool:
         """
-        Returns ``True`` if the validator is active.
+        Return ``True`` if the validator is active.
         """
-        return self.status in VALIDATOR_RECORD_ACTIVE_STATUSES
+        return self.activation_slot <= slot < self.exit_slot
 
     @classmethod
     def get_pending_validator(cls,
```

Comment:
@pipermerriam what are you thoughts on using `get_` naming convention when it is never a retrieval and only a creation? I personally think `create_` or `new_` is clearer in this context.

Is there a previously established convention around this in py-evm?

Reply:
`create_` sounds good to me as it has been used in `eth.rlp.BaseTransaction.create_unsigned_transaction`.